src/components/ciConfig/CICreateDockerfileOption.tsx

```diff
@@ -0,0 +1,428 @@
+                            DropdownIndicator,
+                            Option,
+                        }}
+                        onChange={handleLanguageSelection}
+                        isDisabled={configOverrideView && !allowOverride}
+                    />
+                )}
+                {languageFrameworks?.get(selectedLanguage?.value)?.[0]?.value && (
+                    <>
+                        <div className="h-22 dc__border-right-n1 mr-8 ml-8" />
+                        <span className="fs-13 fw-4 lh-20 cn-7 mr-8">Framework</span>
+                        {configOverrideView && !allowOverride ? (
+                            <span className="fs-13 fw-6 lh-20 cn-9">{selectedFramework?.label}</span>
+                        ) : (
+                            <ReactSelect
+                                tabIndex={3}
+                                options={languageFrameworks?.get(selectedLanguage?.value) || []}
+                                value={selectedFramework}
+                                isSearchable={false}
+                                styles={customStyles}
+                                components={{
+                                    IndicatorSeparator: null,
+                                    DropdownIndicator,
+                                    Option,
+                                }}
+                                onChange={handleFrameworkSelection}
+                                isDisabled={configOverrideView && !allowOverride}
+                            />
+                        )}
+                    </>
+                )}
+                {!editorData?.fetching && editorData?.data !== editorValue && (
+                    <>
+                        <div className="h-22 dc__border-right-n1 mr-8 ml-8" />
+                        <div className="flex left cursor" onClick={resetChanges}>
+                            <div className="icon-dim-12 flex">
+                                <Reset className="icon-dim-12" />
+                            </div>
+                            <span className="ml-4">Reset changes</span>
+                        </div>
+                    </>
+                )}
+            </div>
+        )
+    }
+
+    const handleCopyToClipboard = (e) => {
+        e.stopPropagation()
+        copyToClipboard(editorValue, () => setCopied(true))
+    }
+
+    const handleEditorValueChange = (value: string) => {
+        setEditorValue(value)
+
+        if (templateData && currentCIBuildConfig.dockerBuildConfig?.language) {
+            setCurrentCIBuildConfig({
+                ...currentCIBuildConfig,
+                dockerBuildConfig: {
+                    ...currentCIBuildConfig.dockerBuildConfig,
+                    dockerfileContent: value,
+                },
+            })
+        }
+    }
+
+    const editorData = templateData?.[`${selectedLanguage?.value}-${selectedFramework?.value}`]
```

Comment:
This will produce undefined in case of no selectedLanguage or no selectedFramework

Reply:
Added checks prior getting data from templateData

Reply:
Optional checks are already in place for `editorData`